sympy/matrices/matrices.py

```diff
@@ -3173,6 +3173,122 @@ def _jblock_exponential(b):
         else:
             return type(self)(ret)
 
+    def _eval_matrix_log_jblock(self, principal=True):
+        """Helper function to compute logarithm of a jordan block.
+
+        Examples
+        ========
+
+        >>> from sympy import Symbol, Matrix
+        >>> l = Symbol('lamda')
+
+        A trivial example of 1*1 Jordan block:
+
+        >>> m = Matrix.jordan_block(1, l)
+        >>> m._eval_matrix_log_jblock()
+        Matrix([[log(lamda)]])
+
+        An example of 3*3 Jordan block:
+
+        >>> m = Matrix.jordan_block(3, l)
+        >>> m._eval_matrix_log_jblock()
+        Matrix([
+        [log(lamda),    1/lamda, -1/(2*lamda**2)],
+        [         0, log(lamda),         1/lamda],
+        [         0,          0,      log(lamda)]])
+        """
+        size = self.rows
+        l = self[0, 0]
+
+        if l.is_zero:
+            raise MatrixError(
+                'Could not take logarithm or reciprocal for the given '
+                'eigenvalue {}'.format(l))
+
+        if l.is_negative and principal:
```

Comment:
I see no reason why the principal values of matrix logatrithms could not be defined in the same way as for scalars (taking `pi*I` as the imaginary part for negative eigenvalues).

Reply:
I agree